src/Module/Menu.cpp

```diff
@@ -41,58 +48,57 @@ Menu::Menu(sf::RenderWindow *window, RacePtr race) :
                 close_ = true;
             }
     );
-
-    // every drawable object
-    objects_.push_back(playButton_);
-    objects_.push_back(workshopButton_);
-    objects_.push_back(quitButton_);
-
-    // buttons
-    buttons_.push_back(playButton_);
-    buttons_.push_back(workshopButton_);
-    buttons_.push_back(quitButton_);
+    RegisterButton(quitButton_);
 }
 
-
 int Menu::Run() {
     close_ = false;
     while (!close_) {
         bool mouseClick = false; // info for buttons
 
-        // EVENT handling
-        sf::Event event;
-        while (window_->pollEvent(event)) {
-            // "close_ requested" event: end while loop
-            if (event.type == sf::Event::Closed ||
-                (event.type == sf::Event::KeyPressed && event.key.code == sf::Keyboard::Key::Escape)) {
-                close_ = true;
-            }
-            // handle mouse click
-            if (event.type == sf::Event::MouseButtonPressed && event.mouseButton.button == sf::Mouse::Left) {
-                mouseClick = true;
-            }
-        }
+        HandleEvent(mouseClick);
 
-        /// BUTTON things
-        std::shared_ptr<Button> activeButton = nullptr;
-        sf::Vector2f mousePos = sf::Vector2f(sf::Mouse::getPosition(*window_));
-        // select Hovered button, change style
-        for (auto &button : buttons_) {
-            if (button->ToggleHover(mousePos)) {
-                activeButton = button;
-            }
+        ServeButtons(mouseClick);
+
+        Draw();
+    }
+
+    return 0;
+}
+
+void Menu::HandleEvent(bool &mouseClick) {
+    sf::Event event;
+    while (window_->pollEvent(event)) {
+        // "close_ requested" event: end while loop
```

Comment:
Sam miałem dylemat. Bool mi bardziej pasował, ale za to 0 i 1sa z konwencji sukces/porażka